### DISCUSSION TITLE
Grant external SD-card write permissions as root?

### DISCUSSION THREAD

**Discussion by Wynadorn** [termux/termux-app#2762] [Q&A] [answered]
Is it possible to grant Termux write access to the external sd-card as root?
For example by adding the uid (`u0_a425`) to the `sdcard_rw` group.

When ssh'ing into termux I'd like to avoid using the root user.
I guess it'd be possible to `chmod 777` the entire external sd, though I'd like to avoid it.

What's the propper way to set this up?
Is it even possible?

**Comment by Wynadorn** [comment]
I'm looking to do this because I have my phone's storage mounted though `sshfs` on my desktop.
So it'd be nice to have read/write access on the sd-card.

**Reply by Wynadorn** [reply]
In case anyone else wants to set up something similar in the furture

On my PC in wsl I run `/usr/bin/mnt_termux` which then runs `sudo /etc/profile.d/su/mnt_termux.sh` which runs
```sshfs -v wyn@192.168.1.10:/data/data/com.termux/files/home /mnt/termux -p 8022 -o IdentityFile=/home/wyn/.ssh/id_rsa -o allow_other -o follow_symlinks -o reconnect 2> /dev/null &```

I have this indirect construction so I can run sudo on `mnt_termux.sh` without password, cause `/etc/profile.d/su/` is set to `NOPASSWD` in `/etc/sudoers`

**Comment by agnostic-apollo** [comment, marked as answer]
You can likely use `bindfs` from `root-repo` to remount external sd card with termux uid in termux home `~/external-sd`.

Check `Storage Requirements` section at the end of https://github.com/termux/termux-packages/pull/7227#issuecomment-893022283

**Reply by Wynadorn** [reply]
Very cool, ill give it  ago

**Reply by Wynadorn** [reply]
all done!
`sudo bindfs --map=0/10425:@0/@10425 "/storage/B8D7-217A/" "/data/data/com.termux/files/home/sdcard/"`

such a nerdflex to take a photo and access instantly it under `\\wsl.localhost\Debian\mnt\termux\sdcard\dcim` on my desktop 😁
